# Estimando $\beta$ de los activos

## <font color=blue> Evaluar el curso. </font>
- [En este enlace](https://valoraccion.iteso.mx/)

## <font color=green> Conclusiones de la lectura. </font>

<img style="float: right; margin: 0px 0px 15px 15px;" src="https://upload.wikimedia.org/wikipedia/commons/4/48/Capital_market_line_of_CAPM.svg" width="400px" height="400px" />

En la clase anterior aprendimos 
- ¿qué es el CAPM?; 
- ¿cuáles son los supuestos sobre los que se funda el CAPM?;
- derivamos la fórmula del CAPM; y
- aprendimos como obtener la $\beta$ de un portafolio a partir de la $\beta$ de activos individuales.

En la clase de hoy estudiaremos una forma de estimar los $\beta$ de activos individuales.

**Objetivos:**
- Revisitar riesgo sistemático y no sistemático.
- Estudiar un método para estimar las $\beta$ de los activos.

*Referencia:*
- Notas del curso "Portfolio Selection and Risk Management", Rice University, disponible en Coursera.
- [Notas del curso "Financial Engineering", Columbia University](http://www.columbia.edu/~ks20/FE-Notes/FE-Notes-Sigman.html)
___

## 1. Riesgo sistemático y no sistemático.

Recordamos la fórmula de CAPM:

$$E[r_i]-r_f=\beta_i(E[r_M]-r_f),$$

donde $\beta_i=\frac{\sigma_{M,i}}{\sigma_M^2}$ y $\sigma_{M,i}$ es la covarianza del portafolio de mercado con el activo individual $i$.

Todas las anteriores son variables determinísiticas.

- ¿Qué pasa si usamos el CAPM como un modelo de rendimientos? Es decir,

$$r_i=r_f+\beta_i(r_M-r_f)+\epsilon_i,$$

donde $\epsilon_i$ es un término de error. (error asociado a la aproximación)

<font color=green> Despejando $\epsilon_i$, tenemos que: </font>

- $E[\epsilon_i]=0$, y

    $E[r_f]=E[r_i -r_f -\beta_i(r_M -r_f)]$
    
    $E[r_f]=E[r_i] -E[r_f] -\beta_i(E[r_M] -E[r_f])$
    
    $donde E[r_f] = r_f$
    
    $E[r_f]=0$
    
    
- $cov(\epsilon_i,r_M)=0$.
    
    $cov(\epsilon_i,r_M)=cov(r_i -r_f -\beta_i(r_M -r_f), r_M)$
    
    $cov(\epsilon_i,r_M)=cov(r_i -\beta_i r_M, r_M)$
    
    $cov(\epsilon_i,r_M)=cov(r_i,r_M) -\beta_i cov(r_M -r_M)$
    
    $donde... r_i = r_f + \beta_i(r_m -r_f)+\epsilon_i$
    
    $$sustituyendo...$$
    
    $E[r_i]= E[r_i -r_f -\beta_i(r_M -r_f)]=0$
    
    $var(r_i)=cov(r_i, r_i)$
    
    $var(r_i)=cov(r_f +\beta_i(r_M -r_f)+\epsilon_i, r_f+\beta_i(r_M -r_f)+\epsilon_i)$
    
    $var(r_i)=cov(\beta_i r_M +\epsilon_i , \beta_i r_M + \epsilon_i$
     
    $var(r_i)=\beta_i^2 \sigma_M^2 + 2\beta_i cov(r_M, \epsilon_i)+var(\epsilon_i)$

<font color=green> Ver en el tablero. </font>

Entonces, la varianza del activo $i$ es:

$$\sigma_i^2=\beta_i^2\sigma_M^2+var(\epsilon_i),$$

donde el primer término corresponde al riesgo sistemático (de mercado) y el segundo al riesgo idiosincrático.
___

## 2. Estimando $\beta$ para un activo.

- En el mercado real, la cantidad de activos es ENORME, y tratar de construir el portafolio de mercado sería una tarea grandiosa, pero poco realista para un analista financiero. 
- Por lo tanto, los índices de mercado han sido creados para intentar aproximar el portafolio de mercado.
- Dicho índice es un portafolio más pequeño que el de mercado, construido por los que se consideran los activos más dominantes, y que capturan la esencia de el portafolio de mercado.

- El índice de mercado más conocido es el Standard & Poor’s 500-stock index (S&P), compuesto de 500 activos.
- Un $\beta$ para un activo dado, se puede estimar utilizando el S&P en reemplazo de M, y utilizando datos históricos para ambos rendimientos (el del activo y el del S&P500).
- Por ejemplo, considere un activo $i$ para el cual queremos estimar su $\beta_i$.
- Este estimado se construye usando medias, varianzas y covarianzas muestrales como sigue:
 - Escogemos $N$ rendimientos históricos, tales como los reportados mensualmente hace tres años.
 - Para $k=1,2,\dots,N$, $r_{ik}$ y $r_{S\&Pk}$ denotan el $k-$ésimo valor muestral de los rendimietos.

Entonces

$$\hat{E[r_i]}=\frac{1}{N}\sum_{k=1}^{N}r_{ik}, \text{ y}$$

$$\hat{E[r_{S\&P}]}=\frac{1}{N}\sum_{k=1}^{N}r_{S\&Pk}.$$

Además, la varianza $\sigma_{S\&P}^2$ se estima como

$$\hat{\sigma_{S\&P}^2}=\frac{1}{N-1}\sum_{k=1}^{N}(r_{S\&Pk}-\hat{E[r_{S\&P}]})^2,$$

y la covarianza $\sigma_{S\&P,i}$

$$\hat{\sigma_{S\&P,i}}=\frac{1}{N-1}\sum_{k=1}^{N}(r_{S\&Pk}-\hat{E[r_{S\&P}]})(r_{ik}-\hat{E[r_i]})$$

Finalmente, 

$$\hat{\beta_i}=\frac{\hat{\sigma_{S\&P,i}}}{\hat{\sigma_{S\&P}^2}}.$$

### Ejemplo...

Entrar a yahoo finance, a la información de MSFT, AAPL y ^GSPC.

In [1]:
# Importar paquetes
import pandas as pd
import pandas_datareader.data as web

In [2]:
# Función para descargar precios de cierre ajustados:
def get_adj_closes(tickers, start_date=None, end_date=None, freq='m'):
    # Fecha inicio por defecto (start_date='2010-01-01') y fecha fin por defecto (end_date=today)
    # Descargamos DataFrame con todos los datos
    closes = web.YahooDailyReader(symbols=tickers, start=start_date, end=end_date, interval=freq).read()['Adj Close']
    # Se ordenan los índices de manera ascendente
    closes.sort_index(inplace=True)
    return closes

In [6]:
# Importar datos de AAPL, MSFT, y ^GSPC
names= ['AAPL', 'MSFT','^GSPC']
start_date, end_date= '2016-11-01','2019-11-01'
closes= get_adj_closes(tickers=names, 
                    start_date=start_date, 
                    end_date=end_date)

In [7]:
# Obtener precios mensuales
closes.head()

Symbols           AAPL       MSFT        ^GSPC
Date                                          
2016-11-01  105.036453  56.906445  2198.810059
2016-12-01  110.638626  59.078251  2238.830078
2017-01-01  115.921227  61.464577  2278.870117
2017-02-01  130.861588  60.827587  2363.639893
2017-03-01  137.828171  62.994560  2362.719971

In [8]:
# Obtener rendimientos mensuales
ret = data.pct_change().dropna()
ret.head()

Symbols         AAPL      MSFT     ^GSPC
Date                                    
2016-12-01  0.053336  0.038165  0.018201
2017-01-01  0.047746  0.040393  0.017884
2017-02-01  0.128884 -0.010364  0.037198
2017-03-01  0.053236  0.035625 -0.000389
2017-04-01 -0.000070  0.039478  0.009091

In [9]:
# Obtener matriz de covarianza
covariance= ret.cov()
covariance

Symbols      AAPL      MSFT     ^GSPC
Symbols                              
AAPL     0.006226  0.001684  0.001386
MSFT     0.001684  0.001977  0.001206
^GSPC    0.001386  0.001206  0.001207

In [10]:
# Beta de Microsoft
beta_MSFT= covariance.loc['MSFT', '^GSPC']/covariance.loc['^GSPC','^GSPC']
beta_MSFT

0.9996177982567852

In [11]:
# Beta de Apple
beta_MSFT= covariance.loc['AAPL', '^GSPC']/covariance.loc['^GSPC','^GSPC']
beta_MSFT

1.1480474227080073

## 3. Ejercicios

Su portafolio de inversión consiste de $\$150,000$ invertidos en un solo activo. Suponga que la tasa de interés libre de riesgo es $3\%$. Este activo tiene un rendimiento esperado del $12\%$, y una volatilidad del $40\%$, y el portafolio de mercado tiene un rendimiento esperado del $10\%$ y una volatilidad del $18\%$. Bajo las suposiciones del CAPM, ¿cuál es la volatilidad de un mejor portafolio alternativo que tiene el mismo rendimiento que el activo?

A. $\sigma = 15.52\%$

B. $\sigma = 23.14\%$esta

C. $\sigma = 30.25\%$ 

D. $\sigma = 35.13\%$ 

$$E[r]=r_f+\frac{E[r_M]-r_f}{\sigma_M}\sigma.$$

Necesitamos despejar sigma 

$$\sigma= (\frac{E[r_i]-r_f}{E[r_M]-r_f})\sigma_M$$

In [13]:
rf= 0.03
Erf= 0.12
srf= 0.4
Erm=0.1
srm=0.18

sigma= srm*((Erf-rf)/(Erm-rf))
sigma

0.2314285714285714

Con los mismos datos del ejercicio anterior, y suponiendo que usted está conforme con el riesgo del activo, bajo las suposiciones de CAPM, ¿cuál sería un mejor rendimiento esperado que debería ganar?

A. $E[r] = 13.65\%$

B. $E[r] = 15.30\%$

C. $E[r] = 18.56\%$esta

D. $E[r] = 20.38\%$ 

In [14]:
rf+((Erm-rf)/srm)*srf

0.18555555555555558

Suponga que usted tiene invertidos $\$30,000$ en los siguientes cuatro activos:

| Activo        | Cantidad invertida | $\beta$ |
| ------------- | ------------------ | ------- |
| Activo A      | $\$5,000$          | 0.75    |
| Activo B      | $\$10,000$         | 1.10    |
| Activo C      | $\$8,000$          | 1.36    |
| Activo D      | $\$7,000$          | 1.88    |

La tasa libre de riesgo es del $2\%$ y el rendimiento esperado del portafolio de mercado es del $8\%$. Con base en el CAPM, ¿cuál es la $\beta$ del portafolio?

A. $\beta_P = 0.95$

B. $\beta_P = 1.19$

C. $\beta_P = 1.29$esta

D. $\beta_P = 1.62$

In [18]:
w1, w2, w3, w4= 1/6, 1/3, 4/15, 7/30
b1, b2, b3, b4= 0.75,1.1,1.36,1.88
w1*b1+w2*b2+w3*b3+w4*b4

1.2930000000000001

Considere la siguiente distribución de rendimientos:

| Estado de la economía | Probabilidad       | $R_A$   |
| --------------------- | ------------------ | ------- |
| Depresión             | 0.30               | 0.10    |
| Normal                | 0.50               | 0.15    |
| Expansión             | 0.20               | 0.25    |

Suponiendo que el CAPM es válido, que la volatilidad del portafolio de mercado es $\sigma_M=0.10$ y que la correlación del activo A y el portafolio de mercado es $\rho_{A,M}=0.9$, ¿cuál es el valor de $\beta_A$?

A. $\beta_A=0.25$

B. $\beta_A=0.47$esta

C. $\beta_A=0.55$

D. $\beta_A=1.15$

In [25]:
Er=0.3*0.1+0.5*0.15+0.20*0.25
vol= (0.3*(0.1-Er)**2+0.5*(0.15-Er)**2+0.2*(0.25-Er)**2)**0.5
beta=(0.9*vol)/0.1
Er, vol,beta

(0.155, 0.05220153254455275, 0.46981379290097475)

Un fondo de inversiones con $\beta=0.8$ tiene un rendimiento esperado del $16\%$. Si la tasa libre de riesgo es $4\%$ y el rendimiento esperado del portafolio de mercado es del $13\%$, ¿debería invertir en este fondo?

A. Si. Porque en este momento se encuentra subvaluado y su precio es menor, tambien esta generando mayor rendimiento del esperado. 

B. No.

In [21]:
beta=0.8
Ei=0.16
rf=0.04
Em=0.13

rf+beta*(Em-rf)

0.11199999999999999

Suponga que encontramos un activo el cual, con base en su precio al día de hoy, tiene un rendimiento esperado más bajo que lo que sugiere la línea de mercado de activos. Si el CAPM es válido, ¿cuál(es) de las siguientes proposiciones es (son) verdaderas?

A. Si el activo tiene un rendimiento esperado más bajo que el sugerido por la línea, significa que su precio es muy bajo.

*B. Si el activo tiene un rendimiento esperado más bajo que el sugerido por la línea, significa que su precio es muy alto.*

C. El activo se grafica por encima de la línea.

*D. El activo se grafica bajo la línea.*

# Anuncios parroquiales

## 1. Último quiz la próxima clase.
## 2. Tarea 7 para el viernes 15 de noviembre.

<script>
  $(document).ready(function(){
    $('div.prompt').hide();
    $('div.back-to-top').hide();
    $('nav#menubar').hide();
    $('.breadcrumb').hide();
    $('.hidden-print').hide();
  });
</script>

<footer id="attribution" style="float:right; color:#808080; background:#fff;">
Created with Jupyter by Esteban Jiménez Rodríguez.
</footer>